### ISSUE TITLE
Failsafe with RetryPolicy and CircuitBreaker throws CircuitBreakerOpenException

### ISSUE THREAD

**Issue by omblanco** [jhalterman/failsafe#119]
I am trying to combine a retry policy with the CircuitBreaker pattern  with Failsafe but I get a CircuitBreakerOpenException exception when an  attempt is made with the circuit open and it is interrupted.

The problem is generated by setting a delay for retries less than the circuit's closing time.

How can I control this exception so that the retry policy is not interrupted? I want to do this because I can have several simultaneous instances launched requests to a rest service and that retries are not interrupted. Am I configuring something wrong? Thanks.

My code:

```
public class UnstableApplication {
    private final int MAX_FAILS = 4;
    private AtomicInteger failCount = new AtomicInteger(1);

    public String generateId() throws Exception {
        if (failCount.getAndIncrement() < MAX_FAILS) {
            System.err.printf("UnstableApplication throws SampleException at '%s'\n", ZonedDateTime.now());
            throw new Exception();
        }

        final String id = UUID.randomUUID().toString();
        System.out.printf("UnstableApplication: id '%s' generated at '%s'\n", id, ZonedDateTime.now());

        return id;
    }

}

public class FailsafeExample {

    public static void main(String[] args) throws Exception {

        UnstableApplication app = new UnstableApplication();

        RetryPolicy retryPolicy = new RetryPolicy()
                .retryOn(Exception.class)
                .withDelay(2, TimeUnit.SECONDS)
                .withMaxRetries(5);

        CircuitBreaker breaker = new CircuitBreaker();
        breaker.withFailureThreshold(2);
        breaker.withDelay(5, TimeUnit.SECONDS);
        breaker.withSuccessThreshold(3);
        breaker.onOpen(() -> {
            System.out.println("Circuit breaker is open");
        });

        breaker.onClose(() -> {
            System.out.println("Circuit breaker is close");
        });

        breaker.onHalfOpen(() -> {
            System.out.println("Circuit breaker is half-close");
        }); 

        Failsafe.with(retryPolicy)
        .with(breaker)
        .onFailedAttempt((a, b) -> {
            System.out.println(
                    String.format("Failed with exception: %s, at %s, circuit-breaker state is: %s", 
                            b, ZonedDateTime.now(), breaker.getState()));
        })
        .onSuccess(cxn -> {
            System.out.println("Succcess!");
        })
        .onFailure(cxn -> {
            System.out.println("Failed!");
        })
        .get(new Callable<String>() {
            @Override
            public String call() throws Exception {
                return app.generateId();
            }
        });
    }
}

```

`My result:`

> UnstableApplication throws SampleException at '2017-12-29T16:54:25.264+01:00[Europe/Paris]' Failed with exception: java.lang.Exception, at 2017-12-29T16:54:25.280+01:00[Europe/Paris], circuit-breaker state is: CLOSED UnstableApplication throws SampleException at '2017-12-29T16:54:27.280+01:00[Europe/Paris]' Circuit breaker is open Failed with exception: java.lang.Exception, at 2017-12-29T16:54:27.281+01:00[Europe/Paris], circuit-breaker state is: OPEN Exception in thread "main" net.jodah.failsafe.CircuitBreakerOpenException at net.jodah.failsafe.SyncFailsafe.call(SyncFailsafe.java:136) at net.jodah.failsafe.SyncFailsafe.get(SyncFailsafe.java:56) at com.colobu.rest.vertx.FailsafeExample.main(FailsafeExample.java:57)

**Comment by whiskeysierra** [comment]
> so that the retry policy is not interrupted

What do you mean with *interrupted* exactly?

**Comment by whiskeysierra** [comment]
Looks like it's related to #96.

**Comment by jhalterman** [comment]
Proper support for composable Policies was merged via 5d919d5. This should be release soon as part of 2.0.
